swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -132,14 +171,151 @@ var retrievalSimServiceMap = map[string]simulation.ServiceFunc{
+			if err != nil {
+				return nil, nil, err
+			}
+
+			r := NewRegistry(addr.ID(), delivery, netStore, state.NewInmemoryStore(), &RegistryOptions{
+				Syncing: SyncingDisabled, // disable syncing
+			}, nil)
+
+			cleanup = func() {
+				r.Close()
+				clean()
+			}
+
+			return r, cleanup, nil
+		},
+	},
+	)
+	defer sim.Close()
+
+	log.Info("Initializing test config", "node count", nodeCount)
+
+	conf := &synctestConfig{}
+	//map of discover ID to indexes of chunks expected at that ID
+	conf.idToChunksMap = make(map[enode.ID][]int)
+	//map of overlay address to discover ID
+	conf.addrToIDMap = make(map[string]enode.ID)
+	//array where the generated chunk hashes will be stored
+	conf.hashes = make([]storage.Address, 0)
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancelSimRun()
+
+	filename := fmt.Sprintf("testing/snapshot_%d.json", nodeCount)
+	err := sim.UploadSnapshot(ctx, filename)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Starting simulation")
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		nodeIDs := sim.UpNodeIDs()
+		// first iteration: create addresses
+		for _, n := range nodeIDs {
+			//get the kademlia overlay address from this ID
+			a := n.Bytes()
+			//append it to the array of all overlay addresses
+			conf.addrs = append(conf.addrs, a)
+			//the proximity calculation is on overlay addr,
+			//the p2p/simulations check func triggers on enode.ID,
+			//so we need to know which overlay addr maps to which nodeID
+			conf.addrToIDMap[string(a)] = n
+		}
+
+		// now create random chunks
+		chunks := make([]chunk.Chunk, chunkCount)
 
-The snapshot should have 'streamer' in its service list.
-*/
+		for i := 0; i < chunkCount; i++ {
+			chunks[i] = storage.GenerateRandomChunk(int64(chunkSize))
+			conf.hashes = append(conf.hashes, chunks[i].Address())
+		}
+		log.Debug("random chunks generated, mapping keys to nodes")
+
+		// map addresses to nodes
+		mapKeysToNodes(conf)
+
+		// second iteration: store chunks at the nodes they would be
+		// expected to be
+		log.Debug("storing every chunk at correspondent node store")
+		for _, id := range nodeIDs {
+			// these are the chunks for this node
+			localChunks := conf.idToChunksMap[id]
+			// for every such chunk (which are only indexes)...
+			for _, ch := range localChunks {
+				item, ok := sim.NodeItem(id, bucketKeyStore)
+				if !ok {
+					return fmt.Errorf("Error accessing localstore")
+				}
+				lstore := item.(chunk.Store)
+				// ...get the actual chunk
+				for _, chnk := range chunks {
+					if bytes.Equal(chnk.Address(), conf.hashes[ch]) {
+						// ...and store it in the localstore
+						_, err = lstore.Put(ctx, chunk.ModePutUpload, chnk)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+
+		// now try to retrieve every chunk from every node
+		log.Debug("starting retrieval")
+		cnt := 0
+
+	REPEAT:
+		for {
```

Comment:
so this test suits the old `action, trigger, check` pattern

Reply:
that would also make it concurrent, which is also better for testing realistic scenarios.
I would just allocate chunks to nodes to retrieve them independently, otherwise caching may distort the `pure` nature of this test

Reply:
I am not sure I understand what you mean here and I am also not sure what action I should take on your comments. I have thus currently not changed anything

Reply:
https://github.com/ethersphere/go-ethereum/tree/master/p2p/simulations#testing-framework

an example is https://github.com/ethersphere/go-ethereum/blob/master/swarm/network/simulations/discovery/discovery_test.go#L221

Reply:
Oh I understand of course how the old simulation pattern worked. but I had not understood you actually suggested to migrate this simulation to that setup.

I am against this for a couple of reasons:
* All simulations in the `stream` package run with the new framework, even in this same file there are two other simulations. The sync simulation also runs with the new framework. I really would not like to see just one simulation in a different framework...
* I don't see a much higher value to have to refactor this to the old framework. 
* As I have commented elsewhere in this PR, I don't see value in trying to eliminate caching from retrieval. First, it is intrinsically part of retrieval, and you yourself have made this point several times. We would be putting on this test an onus we don't on any other test nor on production. Second, it would be an incomplete attempt at doing that, as it is currently not possible to thoroughly separate retrieval from syncing as far as I know, so *some* caching would still be happening.